bee-node/bee-plugin/bee-plugin-version-checker/src/lib.rs

```diff
@@ -5,43 +5,167 @@
+                log::info!("On the latest release version ({}).", current_version);
+            }
+        }
+        None => log::warn!("Could not identify the latest release version."),
+    }
+}
+
+/// Returns a list of [`ReleaseInfoBuilder`]s describing all Github releases if the API request and
+/// deserialization was successful. Otherwise, returns `None`.
+async fn get_release_info() -> Option<Vec<ReleaseInfoBuilder>> {
+    let client = reqwest::Client::builder().user_agent("bee").build().unwrap();
+
+    match client
+        .get("https://api.github.com/repos/iotaledger/bee/releases")
+        .send()
+        .await
+        .and_then(|resp| resp.error_for_status())
+    {
+        Ok(resp) => match resp.json::<Vec<ReleaseInfoBuilder>>().await {
+            Ok(releases) => Some(releases),
+            Err(e) => {
+                // Error deserializing response.
+                log::error!("Error deserializing Github API reponse: {}", e);
+                None
+            }
+        },
+        Err(e) => {
+            // Error occured during API request.
+            log::error!("Error performing Github API request: {}", e);
+            None
+        }
+    }
+}
+
+/// Returns the latest release given a vector of [`ReleaseInfoBuilder`]s.
+///
+/// If the current `bee-node` version is a pre-release, this will include later pre-release versions.
+fn latest(release_info: Vec<ReleaseInfoBuilder>, pre_release: bool) -> Option<ReleaseInfo> {
+    release_info
+        .into_iter()
+        .filter_map(|info| info.build(pre_release))
+        .max()
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn full_release() {
+        let current_version = semver::Version::parse("0.2.0").unwrap();
+        let release_info = vec![
+            ReleaseInfoBuilder {
+                html_url: "".to_string(),
+                tag_name: "v0.1.0".to_string(),
+            },
+            ReleaseInfoBuilder {
+                html_url: "".to_string(),
+                tag_name: "v0.2.0".to_string(),
+            },
+            ReleaseInfoBuilder {
+                html_url: "".to_string(),
+                tag_name: "v0.3.0".to_string(),
+            },
+        ];
+
+        let latest = latest(release_info, false).unwrap();
+
+        assert!(current_version < latest.version);
+        assert_eq!(latest.version.to_string(), "0.3.0".to_string());
+    }
+
+    #[test]
+    fn pre_release() {
+        let current_version = semver::Version::parse("0.3.0-rc4").unwrap();
+        let release_info = vec![
+            ReleaseInfoBuilder {
+                html_url: "".to_string(),
+                tag_name: "v0.2.0".to_string(),
+            },
+            ReleaseInfoBuilder {
+                html_url: "".to_string(),
+                tag_name: "v0.3.0-rc4".to_string(),
+            },
+            ReleaseInfoBuilder {
+                html_url: "".to_string(),
+                tag_name: "v0.3.0-rc5".to_string(),
```

Comment:
We have been doing stuff like `v0.3.0-rc5-trigger-docker`, which doesn't really follow SemVer. Should we stop doing that and switch to something that's compatible, or would it make sense to catch stuff like that in the `version-checker-plugin`?

Reply:
It is really non-compliant with semver though ?

Reply:
Also these tags are just tests until we stabilize docker, we won't do that anymore at some point.

Reply:
I don't think there's anything in the semver spec that indicates this is incorrect usage, so it should be fine.

What might be worth considering though is that, with our current pre-release naming `<prefix><number>`, we might run into problems with version precedence later, since dot-separated terms are only compared numerically if they only contain  [numerical values](https://docs.rs/semver/1.0.6/semver/struct.Prerelease.html#total-ordering). From the spec:
> Precedence for two pre-release versions with the same major, minor, and patch version MUST be determined by comparing each dot separated identifier from left to right until a difference is found as follows:
>
>    Identifiers consisting of only digits are compared numerically.
>
>    Identifiers with letters or hyphens are compared lexically in ASCII sort order.

\
So with our current naming system, if there was ever an `rc11` version, the precedence would be wrong: 
```
0.3.0-rc1 < 0.3.0-rc11 < 0.3.0-rc2
```

Maybe it's worth changing to use `<prefix>.<number>` instead? That would keep the order correct:
```
0.3.0-rc.1 < 0.3.0-rc.2 < 0.3.0-rc.11
```

Reply:
(And in the case of the above, I guess we would use `rc.5.trigger-docker`?)

Reply:
Yeah I actually think `rc.*` is the proper way of doing it

Reply:
Yeah, the precedence was where I was getting at. I also like the "dot" notation.